### ISSUE TITLE
Uvicorn throws error on startup in wsgi mode.

### ISSUE THREAD

**Issue by apollo13** [Kludex/uvicorn#235]
I am trying to run a Django app with wsgi mode enabled:
```
uvicorn --wsgi bap_services.wsgi:application       
Started server process [3141]
None - ASGI [1] Initialized {'type': 'lifespan'}
Waiting for application startup.
None - ASGI [1] Started task
None - ASGI [1] Sent {'type': 'lifespan.startup'}
None - ASGI [1] Raised exception
Exception in 'lifespan' protocol
Traceback (most recent call last):
  File "/home/florian/dev/bap/bap-services/.venv/lib/python3.6/site-packages/uvicorn/lifespan.py", line 29, in run
    await self.asgi(self.receive, self.send)
  File "/home/florian/dev/bap/bap-services/.venv/lib/python3.6/site-packages/uvicorn/middleware/message_logger.py", line 59, in __call__
    await self.inner(self.receive, self.send)
  File "/home/florian/dev/bap/bap-services/.venv/lib/python3.6/site-packages/uvicorn/middleware/wsgi.py", line 82, in __call__
    environ = build_environ(self.scope, message, body)
  File "/home/florian/dev/bap/bap-services/.venv/lib/python3.6/site-packages/uvicorn/middleware/wsgi.py", line 12, in build_environ
    "REQUEST_METHOD": scope["method"],
KeyError: 'method'

Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
```
It seems to perform okay, but the keyerror at startup is weird.

**Comment by lovelydinosaur** [comment]
Gotcha. The WSGI middleware ought to:

1. Handle ‘http’ messages.
2. Log errors on ‘websocket’ messages.
3. Ignore anything else, in this case ‘lifespan’

**Comment by lovelydinosaur** [comment]
If `WSGIMiddleware.__call__` performs an `asset scope[‘type’] == ‘http’` then everything ought to start working again.

This is an integration bug that’s been introduced since we added lifespan messaging.

**Comment by lovelydinosaur** [comment]
Believe this to now be resolved in 0.3.15

**Comment by apollo13** [comment]
Thanks, this is what I get now:
```
DEBUG: Using selector: EpollSelector
INFO: Started server process [14158]
DEBUG: None - ASGI [1] Initialized {'type': 'lifespan'}
DEBUG: None - ASGI [1] Raised exception
DEBUG: Lifespan protocol is not recognized by the application
INFO: Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
```
-- all good :+1:
